### PR TITLE
Rework test coverage facility w/ istanbul

### PR DESCRIPTION
**Pull request by saper**
1) Use istanbul instead of jscoverage
2) generate nice HTML report with branches
3) include 'spec.js' tests in the coverage reporting - hits/line ratio on index.js ~300
4) show little coverage summary in the console output
5) do not use spawn and shell to launch mocha and coveralls clients - wire reporters directly via JavaScript

### CODE REVIEW

**Comment by saper** [comment]
Well, it looks like travis-ci now testing [with io.js 3.0.0](https://travis-ci.org/sass/node-sass/jobs/74660239) as the "latest".

**Comment by saper** [comment]
@xzyfer @am11 what do you think about this? I like not having to invoke shell to do coverage testing (although the resulting JS file is not the simplest).

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
   "devDependencies": {
     "coveralls": "^2.11.2",
     "cross-spawn": "^1.0.3",
-    "jscoverage": "^0.5.9",
+    "istanbul": "^0.3.13",
```
Comment (am11) [inline package.json]: nit: latest version is `0.3.18`.

**Comment by am11** [comment]
LGTM. Lets see if CI agrees. :)

> Well, it looks like travis-ci now testing with io.js 3.0.0 as the "latest".

As per [docs](http://www.appveyor.com/docs/lang/nodejs-iojs), seems like we need to change this line: https://github.com/sass/node-sass/blob/f95af4/appveyor.yml#L12 to `- nodejs_version: 3` in order to have same affect in AppVeyor.

**Comment by xzyfer** [comment]
I'm fine with this.

**Pull request merged**